components/TARGET_PSA/inc/psa/client.h

```diff
@@ -17,7 +17,43 @@
 
 #if defined(TARGET_TFM)
 #include "interface/include/psa_client.h"
-#else
+#elif defined(TARGET_MBED_SPM)
 #include "TARGET_MBED_SPM/psa_defs.h"
 #include "TARGET_MBED_SPM/spm_client.h"
+#else
+
+#ifndef __MBED_OS_DEFAULT_PSA_CLIENT_API_H__
+#define __MBED_OS_DEFAULT_PSA_CLIENT_API_H__
+
+#if !defined(UINT32_MAX)
+#define UINT32_MAX ((uint32_t)-1)
```

Comment:
Why do we duplicating the header file?
Please consider moving spm_client.h and psa_defs.h to this folder instead.

Reply:
spm_client.h and psa_defs.h contains more defines and definitions that are not relevant for non-PSA builds (example K64F)

this "default" implementation holds only what the spec requires.

Reply:
I do not see this default implementation gets removed from the original files. Thus it is a duplication.
Please keep in mind that our divergence from TF-M sources ( spec version) is temporal while single core API compliance is not. Eventually I would not want to duplicate the header files and would prefer adopting TF-M's version.
Please consider moving these files to a common place instead of simbols cherry picking.